src/Illuminate/Database/Eloquent/Relations/Relation.php

```diff
@@ -501,6 +501,17 @@ public static function getMorphedModel($alias)
         return static::$morphMap[$alias] ?? null;
     }
 
+    /**
+     * Get the alias associated with a custom polymorphic class.
+     *
+     * @param  string  $className
+     * @return int|string|null
+     */
+    public static function getMorphAlias(string $className)
+    {
+        return array_flip(static::$morphMap)[$className] ?? null;
```

Comment:
Shouldn't this have been ```array_search($className, static::$morphMap) ?? $className```

Reply:
Fixed: https://github.com/laravel/framework/pull/51845

Reply:
Thanks @calebdw for improving this.